controllers/user/UserController.php

```diff
@@ -98,17 +107,92 @@ public function actionCreate(){
     }
 
     /**
-     * Authenticate the user by username and password
-     * If, the credentials are correct then token will be generated by JWT
+     * Store the user details like email, token in user_invite table
+     * Generate six digit random number code
+     * @param $email
+     * @return array|false
+     * @throws \Throwable
+     * @throws \yii\db\StaleObjectException
+     */
+    public function inviteUser($email){
+        $invite = new Invite();
+        $invite->scenario = 'invite';
+
+        //If the site in maintenance mode
+        if (Yii::$app->settings->get('maintenanceMode')) {
+            Yii::$app->getView()->warn(ControllerAccess::getMaintenanceModeWarningText());
+            return false;
+        }
+
+        $invite->source = Invite::SOURCE_SELF;
+        $invite->language = Yii::$app->language;
+        $invite->email = $email;
+
+        // Delete existing invite for e-mail - but reuse token
+        $existingInvite = Invite::findOne(['email' => $email]);
+        if ($existingInvite !== null) {
+            $invite->token = $existingInvite->token;
+            $existingInvite->delete();
+        }
+
+        $invite->token = $this->generateCode();
+        $checkCodeAlredyExist = Invite::find()->where(['token'=>$invite->token])->one();
+
+        if($checkCodeAlredyExist!=null){
+            $invite->token = $this->generateCode();
+        }
+
+        //We set validation false because we are not fill recaptcha value in API
+        if($invite->save(false)){
+            $this->sendMail($invite);
+            return $this->returnSuccess("Registration successfully done, Please check your email, Code is send to your email address ".$email,200);
+        }
+        return $this->returnError(400,$invite->errors);
+
+    }
+
+    public function generateCode()
+    {
+        return random_int(100000, 999999);
+    }
+
+    /**
+     * Send code to user email address
+     * @param $invite
+     * @return void
+     * @throws \yii\base\Exception
+     * @throws \yii\base\InvalidConfigException
+     */
+    public function sendMail($invite)
+    {
+
+        /** @var Module $module */
+        $module = Yii::$app->moduleManager->getModule('user');
+
+        // User requested registration link by its self
+        if ($invite->source == Invite::SOURCE_SELF) {
+            $mail = Yii::$app->mailer->compose([
+                'html' => '@humhub/modules/smartVillageAPI/views/email_template/UserInviteSelf',
+                'text' => '@humhub/modules/smartVillageAPI/views/email_template/plain_text/UserInviteSelf'
+            ], [
+                'token' => $invite->token,
+            ]);
+            $mail->setTo($invite->email);
+            $mail->setSubject(Yii::t('UserModule.base', 'Welcome to %appName%', ['%appName%' => Yii::$app->name]));
```

Comment:
the text is the subject of the mail, right? so i can edit this to some German one before merging, right?